src/ReactiveDomain.Policy/PolicyDispatcher.cs

```diff
@@ -0,0 +1,116 @@
+﻿using ReactiveDomain.Messaging;
+using ReactiveDomain.Messaging.Bus;
+using ReactiveDomain.Util;
+using System;
+
+namespace ReactiveDomain.Policy
+{
+    public class PolicyDispatcher : IDispatcher
+    {
+        private readonly IDispatcher _dispatcher;
+        private readonly Func<Type, bool> hasPermission;
+        public bool Idle => _dispatcher.Idle;
+        public string Name => _dispatcher.Name;
+
+        public PolicyDispatcher(IDispatcher dispatcher, Func<Type, bool> hasPermission)
```

Comment:
Do we want/need an overload of this that takes a `Func<string, bool>` as input since we support getting permissions by `Type` or by `string` elsewhere?

Reply:
I don't want to allow the slowness of string based compares in the dispatcher.
Also string based permissions that are type names will be resolved to types when `TryResolveType` is called.
This is intended to be part of deserialization from the authorization/policy storage system.
This looks up the types in the same `MessageTypeHierarchy` used by the dispatcher.

Reply:
Test added to clarify